internal/store/bundle_test.go

```diff
@@ -290,3 +292,179 @@ func TestRemove(t *testing.T) {
+		Name          string
+		refOrID       string
+		ExpectedError string
+		ExpectedRef   reference.Reference
+	}{
+		{
+			Name:          "Long Id",
+			refOrID:       id.String(),
+			ExpectedError: "",
+			ExpectedRef:   id,
+		},
+		{
+			Name:          "Short Id",
+			refOrID:       id.String()[0:8],
+			ExpectedError: "",
+			ExpectedRef:   id,
+		},
+		{
+			Name:          "Tagged Ref",
+			refOrID:       "my-repo/a-bundle:my-tag",
+			ExpectedError: "",
+			ExpectedRef:   ref,
+		},
+		{
+			Name:          "docker.io_library repository Tagged Ref",
+			refOrID:       "a-bundle:my-tag",
+			ExpectedError: "",
+			ExpectedRef:   dockerIoRef,
+		},
+		{
+			Name:          "Unknown Tag",
+			refOrID:       "other-repo/a-bundle:other-tag",
+			ExpectedError: "docker.io/other-repo/a-bundle:other-tag: reference not found",
+			ExpectedRef:   nil,
+		},
+		{
+			Name:          "Unknown ID",
+			refOrID:       "b4fcc3af16804e29d977918a3a322daf1eb6ab2992c3cc7cbfeae8c3d6ede8af",
+			ExpectedError: "b4fcc3af16804e29d977918a3a322daf1eb6ab2992c3cc7cbfeae8c3d6ede8af: reference not found",
+			ExpectedRef:   nil,
+		},
+		{
+			Name:          "Unknown short ID",
+			refOrID:       "b4fcc3af",
+			ExpectedError: "docker.io/library/b4fcc3af:latest: reference not found",
+			ExpectedRef:   nil,
+		},
+	} {
+		t.Run(tc.Name, func(t *testing.T) {
+			fmt.Println(tc.refOrID)
+			ref, err := bundleStore.LookUp(tc.refOrID)
+
+			if tc.ExpectedError != "" {
+				assert.Equal(t, err.Error(), tc.ExpectedError)
+			} else {
+				assert.NilError(t, err)
+			}
+
+			if tc.ExpectedRef != nil {
+				assert.Equal(t, ref, tc.ExpectedRef)
+			}
+		})
+	}
+}
+
+func TestScanBundles(t *testing.T) {
+	dockerConfigDir := fs.NewDir(t, t.Name(), fs.WithMode(0755))
+	//defer dockerConfigDir.Remove()
+
+	bndl1 := &bundle.Bundle{Name: "bundle-1"}
+	id1, err := FromBundle(bndl1)
+	assert.NilError(t, err)
+	dir1 := dockerConfigDir.Join("app", "bundles", "_ids", id1.String())
+	assert.NilError(t, os.MkdirAll(dir1, 0755))
+	assert.NilError(t, ioutil.WriteFile(filepath.Join(dir1, "bundle.json"), []byte(`{"name": "bundle-1"}`), 0644))
+
+	bndl2 := &bundle.Bundle{Name: "bundle-2"}
+	id2, err := FromBundle(bndl2)
+	assert.NilError(t, err)
+	dir2 := dockerConfigDir.Join("app", "bundles", "_ids", id2.String())
+	assert.NilError(t, os.MkdirAll(dir2, 0755))
+	assert.NilError(t, ioutil.WriteFile(filepath.Join(dir2, "bundle.json"), []byte(`{"name": "bundle-2"}`), 0644))
+	dir2 = dockerConfigDir.Join("app", "bundles", "docker.io", "my-repo", "my-bundle", "_tags", "my-tag")
+	assert.NilError(t, os.MkdirAll(dir2, 0755))
+	assert.NilError(t, ioutil.WriteFile(filepath.Join(dir2, "bundle.json"), []byte(`{"name": "bundle-2"}`), 0644))
+
+	appstore, err := NewApplicationStore(dockerConfigDir.Path())
+	assert.NilError(t, err)
+	bundleStore, err := appstore.BundleStore()
+	assert.NilError(t, err)
+
+	refs, err := bundleStore.List()
+	assert.NilError(t, err)
+	assert.Equal(t, len(refs), 3)
```

Comment:
Maybe a `DeepEqual` would be better here, so we know what is returned by the `List` function?

Reply:
Fixed by comparing expected reference strings